main.go

```diff
@@ -26,9 +31,19 @@ var (
 	metricTemplate    = kingpin.Flag("metric-template", "The template that will form a new metric namespace.").Default("{{.Space}}.{{.App}}.{{.Instance}}.{{.Metric}}").OverrideDefaultFromEnvar("METRIC_TEMPLATE").String()
 )
 
+func normalizePrefix(prefix string) string {
+	prefix = strings.TrimRight(strings.TrimSpace(prefix), ".")
+	if prefix == "" {
```

Comment:
I think it's completely valid if you don't want a prefix at all. Also there can be tenants who depend on this.

Reply:
I agree that being able to set a blank prefix might be something useful, however the previous behaviour (a result of the kingpin flag passing) was that `STATSD_PREFIX=""` or `--statsd-prefix=""` would result in the default value (`mycf.`) being used.

I didn't want to introduce a breaking change in this PR so the old behaviour was kept.

Reply:
I just tested this and if I set ```--statsd-prefix ""``` then the statsdPrefix variable will be an empty string.

Reply:
hmmm I was testing using the `STATSD_PREFIX=""` environment variable ... which gives the behaviour I described.

But the cli arg can overwrite it as you describe.

I suspect tenants will be using the env var for configuration (since it's a cf app), but  I'm not really sure what the best move is:

1) Leave as-is: which alters behaviour for anyone using `--statsd-prefix=""`
2) Remove the default to allow `STATSD_PREFIX=""` which alters behaviour if anyone has STATSD_PREFIX=""

?

Reply:
I think we should not change the behaviour - at least not in this story (if we ever want to). Simply we should only add the missing "." if statsdPrefix is not empty.

Reply:
Also with your change there will be a way to make the prefix empty even with using the STATSD_PREFIX env var, by passing "." as a value.

Reply:
OK Done.

* `--statsd-prefix=""` works as before resulting in no prefix
* `STATSD_PREFIX=""` works as before as uses the default `mycf` prefix
* `STATSD_PREFIX="."` or `STATSD_PREFIX=" "` is a secret way to set no prefix via env var :)
* `STATSD_PREFIX="withoutadot"` works as expected
* `STATSD_PREFIX="withadot."` works as expected